scripts/prepare_heat_demand.py

```diff
@@ -2,15 +2,441 @@
+        )
+    )
+
+    # Calculate sfh (efh: Einfamilienhaus) heat load
+    heat_load_consumer["efh" + "_" + carrier] = bdew.HeatBuilding(
+        heat_load_consumer.index,
+        holidays=holidays,
+        temperature=temperature,
+        shlp_type="EFH",
+        building_class=building_class,
+        wind_class=0,
+        annual_heat_demand=yearly_demands["efh" + "_" + carrier][0],
+        name="EFH",
+        ww_incl=True,
+    ).get_bdew_profile()
+
+    # Calculate mfh (mfh: Mehrfamilienhaus) heat load
+    heat_load_consumer["mfh" + "_" + carrier] = bdew.HeatBuilding(
+        heat_load_consumer.index,
+        holidays=holidays,
+        temperature=temperature,
+        shlp_type="MFH",
+        building_class=building_class,
+        wind_class=0,
+        annual_heat_demand=yearly_demands["mfh" + "_" + carrier][0],
+        name="MFH",
+        ww_incl=True,
+    ).get_bdew_profile()
+
+    # Calculate industry, trade, service (ghd: Gewerbe, Handel, Dienstleistung)
+    # heat load
+    heat_load_consumer["ghd" + "_" + carrier] = bdew.HeatBuilding(
+        heat_load_consumer.index,
+        holidays=holidays,
+        temperature=temperature,
+        shlp_type="ghd",
+        wind_class=0,
+        annual_heat_demand=yearly_demands["ghd" + "_" + carrier][0],
+        name="ghd",
+        ww_incl=True,
+    ).get_bdew_profile()
+
+    # Calculate total heat load in year
+    heat_load_total[carrier + "-demand-profile"] = heat_load_consumer.sum(axis=1)
+
+    # Normalize heat load profile
+    heat_load_total[carrier + "-demand-profile"] = np.divide(
+        heat_load_total[carrier + "-demand-profile"],
+        heat_load_total[carrier + "-demand-profile"].sum(),
+    )
+
+    return heat_load_total
+
+
+def postprocess_data(heat_load_postprocessed, heat_load_year, region, scenario, unit):
+    """
+    This function stacks time series of heat load profile and addes it to result DataFrame
+
+    Parameters
+    ----------
+    heat_load_postprocessed : pd.Dataframe
+        Empty Dataframe as result DataFrame
+    heat_load_year : pd.Dataframe
+        DataFrame with total normalized heat load in year to be processed
+    region : str
+        Region (eg. Brandenburg)
+    scenario : str
+        Scenario e.g. "base
+    unit : str
+        Unit of total demands (eg. GWh)
+
+    Returns
+    -------
+    heat_load_postprocessed : pd.DataFrame
+         DataFrame that contains stacked heat load profile
+
+    """
+    # Stack time series with total heat load in year
+    heat_load_year_stacked = dp.stack_timeseries(heat_load_year)
+
+    heat_load_year_stacked["region"] = region
+    heat_load_year_stacked["scenario"] = scenario
+    heat_load_year_stacked["var_unit"] = unit[0]
+
+    # Append stacked heat load of year to stacked time series with total heat loads
+    heat_load_postprocessed = pd.concat(
+        [heat_load_postprocessed, heat_load_year_stacked], ignore_index=True
+    )
```

Comment:
Here, I get a future warning because you do not specify if you want to sort or not. Should it be sorted?

Reply:
Done with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/eb699b526ea60ba9e8b3c1b16b511e7dfe312384.